### ISSUE TITLE
WebDAV: Displays incorrect names via Explorer

### ISSUE THREAD

**Issue by Gavrilov-Ivan** [jmix-framework/jmix#1540]
Create webdav collection via Jmix UI using name with whitespace
Connect to Jmix webdav storage via Windows Explorer (add network drive)
Whitespaces are replaced by '+'
Further operations with this collection cause errors because of incorrect name sent back to application.

Also reproduced on WinSCP.

Testing:

- Check operations for both files and folders
- Operations:
  - Create
  - Edit (for files)
  - Rename (for folders, file renaming is not supported)
  - Copy
  - Move from one folder to another
  - Remove
- Naming
  - With & without whitespaces
  - Latin & Cyrillic

**Comment by rusiaikinat** [comment]
case 1
TC
Create file in folder or rename when creating a file
Rename file -> There was a message about a lack of rights to change

AR
1. Now this file cannot be opened from a network drive, but the file can be downloaded from the application.
2. The folder with the modified file cannot be moved

case 2
A folder containing a `+` sign in the name (w+w):
1. Cannot be renamed
2. Cannot be moved to another folder
3. Cannot be removed

case 3
A file named `w ф .txt` from the parent directory is moved to the folder as a folder, not a file

**Comment by Gavrilov-Ivan** [comment]
Fix is commited to release 1.5.
Please perform full retest.

**Comment by rusiaikinat** [comment]
case 1
TC
Create New folder (Новая папка)
Rename folder (Новая папка + _)
Copy folder, paste folder in root directory -> Новая папка + _ — копия
Move the copied folder (Новая папка + _ — копия) to an existing one (Новая папка + _)

AR
"не удается выполнить чтение с файла или диска"
Сannot move a folder to a folder containing in the name +

Note: a folder containing a + in the name can be moved to a folder without this symbol

case 2
files can be inserted into the folder containing the + in the name only by copying

**Comment by Gavrilov-Ivan** [comment]
Changes have been reverted. Will be included in next fix-release.

**Comment by rusiaikinat** [comment]
Reopened
1.
Create 3 folders:
`folder`
`Новая папка`
`Новая папка (2)`

2. Rename folder `Новая папка (2)` -> `Новая папка_2` - it's fine
 2.1 Rename folder `folder` -> `folder2` - **ERRORE** The folder has not been renamed
3. Delete `Новая папка` - it's fine
 3.1 Create folder 'folder3' and then delete it - **ERRORE** The folder was deleted, but after updating the window, it reappeared
4. Move `folder` or `folder2` in `Новая папка_2` - **ERRORE** "Unable to read from file or disk". the behavior is similar if the folder is cut and pasted into another folder.  If the folder contains Cyrillic letters, there are no problems with moving

**Comment by rusiaikinat** [comment]
> Reopened 1. Create 3 folders: `folder` `Новая папка` `Новая папка (2)`
> 
> 2. Rename folder `Новая папка (2)` -> `Новая папка_2` - it's fine
>    2.1 Rename folder `folder` -> `folder2` - **ERRORE** The folder has not been renamed
> 3. Delete `Новая папка` - it's fine
>    3.1 Create folder 'folder3' and then delete it - **ERRORE** The folder was deleted, but after updating the window, it reappeared
> 4. Move `folder` or `folder2` in `Новая папка_2` - **ERRORE** "Unable to read from file or disk". the behavior is similar if the folder is cut and pasted into another folder.  If the folder contains Cyrillic letters, there are no problems with moving

reproduced on hsql

**Case 2**
Mssql12

Create collection in project `12345wwwППП` and collection `12345ПППwww`
![image](https://github.com/jmix-framework/jmix/assets/114649487/a5400c5b-234c-4b21-bac8-81fbd09130b8)

AR
Not all characters in the name are displayed

**Comment by Gavrilov-Ivan** [comment]
**Case 1 (on HSQLDB)**: reproduced also on 1.5.2 before this fix - doesn't relate to this issue. Looks like a specific database bug with clob field.
Due to HSQLDB is not a production DB - decided to skip it.

**Case 2 (MSSQL)**: it doesn't relate to this issue - MSSQL databases created in [cuba-db.haulmont.com](http://cuba-db.haulmont.com/) doesn't support cyrillic by default because of collation. So the project can't use cyrillic anywhere, not only in Webdav.
This can be fixed with manually created database (with SQL server management studio) with proper collation (set up on db creation).
I installed MSSQL locally, create database with `Cyrillic_General_100_CI_AI` collation - it works.

Found another issue: searching for child documents can get an extra data. Affects collection renaming and removal. Can be reproduced in Jmix UI.

1. Create collection `dir1` in root.
2. Create collection `dir2` inside dir1.
3. Create collection `dir_` inside root.
4. Remove collection `dir_`.
5. Go inside `dir1` and check `dir2` - it's also removed

Search is performed by `LIKE` operation. Underscore and percent are wildcard symbols for this operation. 
Fixed via escaping.
